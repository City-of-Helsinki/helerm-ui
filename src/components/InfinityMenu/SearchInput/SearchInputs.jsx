/* eslint-disable react/forbid-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Select from 'react-select';

import SearchInput from './SearchInput';

export const FILTER_CONDITION_OPTIONS = [
  { value: 'and', label: 'JA' },
  { value: 'or', label: 'TAI' },
];

const SearchInputs = ({
  headerProps,
  isDetailSearch,
  searchInputs,
  filterCondition,
  addSearchInput,
  setSearchInput,
  removeSearchInput,
  onFilterConditionChange,
}) =>
  searchInputs.map((input, index) => (
    <div
      // eslint-disable-next-line react/no-array-index-key
      key={index}
      className={classNames({
        'col-xs-12 filters filters-detail-search-input': isDetailSearch,
        'col-sm-6': !isDetailSearch,
      })}
    >
      <SearchInput
        {...headerProps}
        searchInput={input}
        placeholder='Etsi...'
        setSearchInput={(event) => setSearchInput(index, event.target.value)}
      />
      {isDetailSearch && (
        <div className='filters-detail-search-input-buttons'>
          {index + 1 < searchInputs.length && (
            <Select
              className='Select'
              autoBlur
              isDisabled={index > 0}
              placeholder='Ehto'
              value={FILTER_CONDITION_OPTIONS.find(({ value }) => value === filterCondition)}
              isClearable={false}
              options={FILTER_CONDITION_OPTIONS}
              onChange={onFilterConditionChange}
            />
          )}
          <button
            type='button'
            className='btn btn-info btn-sm'
            onClick={() => removeSearchInput(index)}
            title='Poista hakuehto'
            aria-label='Poista hakuehto'
          >
            <span className='fa-solid fa-minus' aria-hidden='true' />
          </button>
          {index + 1 === searchInputs.length && (
            <button
              type='button'
              className='btn btn-info btn-sm'
              onClick={addSearchInput}
              title='Lis채채 hakuehto'
              aria-label='Lis채채 hakuehto'
            >
              <span className='fa-solid fa-plus' aria-hidden='true' />
            </button>
          )}
        </div>
      )}
    </div>
  ));

SearchInputs.propTypes = {
  headerProps: PropTypes.object,
  isDetailSearch: PropTypes.bool,
  searchInputs: PropTypes.array.isRequired,
  filterCondition: PropTypes.string.isRequired,
  addSearchInput: PropTypes.func.isRequired,
  setSearchInput: PropTypes.func.isRequired,
  removeSearchInput: PropTypes.func.isRequired,
  onFilterConditionChange: PropTypes.func.isRequired,
};

export default SearchInputs;
