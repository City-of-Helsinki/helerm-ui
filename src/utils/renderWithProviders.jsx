/* eslint-disable react/prop-types */
import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { LoginProvider } from 'hds-react';

import { initialState as navigationInitialState } from '../store/reducers/navigation';
import { initialState as validationInitialState } from '../store/reducers/validation';
import { initialState as routerInitialState } from '../store/reducers/router';
import { initialState as selectedTOSInitialState } from '../store/reducers/tos-toolkit/main';
import { initialState as classificationInitialState } from '../store/reducers/classification';
import { initialState as userInitialState } from '../store/reducers/user';
import { initialState as uiInitialState } from '../store/reducers/ui';
import { initialState as bulkInitialState } from '../store/reducers/bulk';
import { initialState as searchInitialState } from '../store/reducers/search';
import storeCreator from '../store/createStore';

export const storeDefaultState = {
  navigation: navigationInitialState,
  validation: validationInitialState,
  router: routerInitialState,
  selectedTOS: selectedTOSInitialState,
  classification: classificationInitialState,
  user: userInitialState,
  ui: uiInitialState,
  bulk: bulkInitialState,
  search: searchInitialState,
};

const renderWithProviders = (
  ui,
  { preloadedState = storeDefaultState, store = storeCreator(preloadedState), ...renderOptions } = {},
) => {
  const Wrapper = ({ children }) => (
    <LoginProvider>
      <Provider store={store}>{children}</Provider>
    </LoginProvider>
  );

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

export default renderWithProviders;
