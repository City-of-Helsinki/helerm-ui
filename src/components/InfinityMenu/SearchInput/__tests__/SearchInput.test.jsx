import { createBrowserHistory } from 'history';
import React from 'react';
import { Router } from 'react-router-dom';
import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import SearchInput from '../SearchInput';
import renderWithProviders from '../../../../utils/renderWithProviders';

const renderComponent = (mockFunction = vi.fn()) => {
  const history = createBrowserHistory();

  return renderWithProviders(
    <Router history={history}>
      <SearchInput placeholder='' searchInput='' setSearchInput={mockFunction} />
    </Router>,
    { history },
  );
};

describe('<SearchInput />', () => {
  it('renders correctly', () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });

  it('triggers setSearchInput', async () => {
    const mockSetSearchInput = vi.fn();

    renderComponent(mockSetSearchInput);

    const user = userEvent.setup();
    const input = screen.getByRole('searchbox');

    await user.type(input, 'Test');

    expect(mockSetSearchInput).toHaveBeenCalled();
  });
});
