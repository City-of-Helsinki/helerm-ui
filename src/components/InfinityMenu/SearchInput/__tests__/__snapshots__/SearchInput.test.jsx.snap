// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<SearchInput /> > renders correctly 1`] = `
<div>
  <input
    class="react-infinity-menu-default-search-input"
    placeholder=""
    type="search"
    value=""
  />
</div>
`;
